src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -533,6 +533,7 @@ private function validateAndDenormalize(array $types, string $currentClass, stri
 
                     if (\count($collectionKeyType = $type->getCollectionKeyTypes()) > 0) {
                         [$context['key_type']] = $collectionKeyType;
```

Comment:
Not sure what to do about the `key_type` option, should it be deprecated in 6.4?

Reply:
from what I understand, this was broken already for union types so :shrug:

Reply:
Done